e-commerce/src/pages/catalog/catalog.tsx

```diff
@@ -1,12 +1,216 @@
+        const productsProps = formattedDataForCardInCategory(productsAllGet);
+        setProductCardProps(productsProps);
+      } catch {
+        setError('Failed to fetch products. Please try again later.');
+      } finally {
+        setLoading(false);
+      }
+    };
+    const categories = async () => {
+      const categoriesAllGet: ICategoriesResponse = await getCategories();
+      const categoriesAllData = formattedDataForCategory(categoriesAllGet);
+      setCategoryProps(categoriesAllData);
+    };
+    products();
+    categories();
+  }, []);
+
+  if (loading) {
+    return <Spinner />;
+  }
+
+  if (error) {
+    return <div>{error}</div>;
+  }
+
   return (
-    <>
-      <h2 className="catalog free-page">Catalog</h2>
-      <WetFloor />
-    </>
+    <section className="catalog">
+      <h2 className="catalog-title">Catalog</h2>
+      <div className="catalog-wrapper">
+        <aside className="categories">
+          {categoryProps.map((category) => (
+            <Category
+              key={category.id}
+              name={category.name}
+              id={category.id}
+              onClick={handlerCategoryChoose}
+              isCurrent={category.id === categoryId}
+            />
+          ))}
+          <FilterCatalog getProductsFilter={getProductsFilter} />
+        </aside>
+        <div className="catalog-products">
+          <div className="catalog-sort">
+            <label htmlFor="site-search" className="search-label">
+              <input
+                type="search"
+                id="site-search"
+                name="search"
+                placeholder="Search..."
+                value={searchQuery}
+                onChange={handleChangeInputSearch}
+                className="search-input"
+              />
+              <button
+                className="search-btn"
+                type="button"
+                onClick={handleSearchPanel}
+              >
+                Search
+              </button>
+            </label>
+            <select
+              id="sorted-products"
+              onChange={handleSortChange}
+              className="select-input"
+            >
+              <option value={SortField.Default}>Sort by</option>
+              <option value={SortField.PriceAsc}>Lowest price</option>
+              <option value={SortField.PriceDesc}>Highest price</option>
+              <option value={SortField.CreatedAt}>What&apos;s New</option>
+              <option value={SortField.NameAsc}>Name (A To Z)</option>
+              <option value={SortField.NameDesc}>Name (Z To A)</option>
+            </select>
+          </div>
+          <div className="catalog-product" id="catalog-product">
+            {messageError
+              ? 'Product not found'
+              : productCardProps.map((product) => (
+                  <ProductCard
+                    {...product}
+                    key={product.id}
+                    onClick={handlerProductChoose}
```

Comment:
вообще, еще есть компонент Link, но мб тут его не надо использовать, или надо использовать внутри самого компонента карточки продукта
```suggestion
                    onClick={() => navigate(`/product/${product.id}`)}
```